curie/nodes/base_node.py

```diff
@@ -127,53 +135,25 @@ def Node(state: self.State):
                 content=system_prompt,
             )
 
-            # can filter out early tool messages from state["messages"]
-            # need to retain the plan messages from state["messages"]
-            # remove the duplicated human messages from state["messages"]
-            
-            # TODO: check for high similarity between messages, repeatly summarize the messages
-            # If there are too many messages, prune older ToolMessages to avoid context overflow
             messages = state["messages"]
-            # unique_msg_contents = {} # content -> index
-            if len(messages) > 50:
-                # Keep track of tool messages to potentially remove
-                tool_messages = []
-                to_remove = set()
+            if self.memory_manager_enabled:
+                # Use memory manager to get and prune messages
+                original_messages = state["messages"]
+                original_token_count = count_tokens(original_messages)
+
+                self.memory_manager.replace(self.plan_namespace, original_messages)
+                self.memory_manager.prune_and_summarize(self.plan_namespace)
```

Comment:
Will this be triggered as long as the context length exceeds? Can we cache the summary?